### DISCUSSION TITLE
react-native-windows for unistyles

### DISCUSSION THREAD

**Discussion by jpudysz** [microsoft/react-native-windows#12673] [General]
Hello 👋,

Recently, I added support for react-native-windows to [react-native-unistyles](https://reactnativeunistyles.vercel.app/). Unistyles is a library that offers a stylesheet with superpowers, incorporating custom C++ code and its own event system, separate from react-native.

I would appreciate a brief code review from someone more experienced, as I have some reservations about requesting certain information, such as window dimensions and color scheme, on the UI thread with future and promise. Can it be done differently?

Here is the file that uses JSI to inject the Unistyles object into the global scope. Any comments or feedback would be greatly appreciated!

[ReactNativeUnistyles.h on GitHub](https://github.com/jpudysz/react-native-unistyles/blob/main/windows/ReactNativeUnistyles/ReactNativeUnistyles.h)

**Comment by Saadnajmi** [comment]
Wow, React Native Unistyles on React Native macOS _and_ React Native Windows! The future is bright :D
